### PR TITLE
Add Aux input, temperature reads, fix battery voltage read, fixes

### PR DESCRIPTION
**Pull request by dvosully**
This PR adds the following capabilities:

- Read the Aux Input pin
- Read the temperature using the TEMP0 register (one point mode)
- Read the temperature using the TEMP1 register (two point mode)
- Enable PENIRQ pin option. ~~(Note: Can't use this in the driver to check for touch, but can enable the output from the chip)~~
- Enable Vref ON option. (Leaves the Vref permanently on to reduce transactions when reading battery/aux/temperature

It also fixes the following issues:

- The battery voltage value wasn't being right-shifted by 3 so was 8 times too large.
- Vref is approx 2.507V, not 2.500V (you need to look at the internal Vref vs temperature graph in the datasheet to find that, between 5C and 40C it's above 2.507V, and still above 2.505V between -25C and 65C, which is basically the normal range)
- Fixes co-ordinate averaging.
  - The bug:
    - If the screen press is released while the averaging loop is being run, some co-ordinates with value 0 get introduced into the average.
    - This means that instead of the averaged coordinate value being the press location, it is 'torn' between the press location and the corner of the screen (coordinate 0,0)
    - Because this happens on a press release, it results in a 'click' action in a completely different location than the press.
    - This is more likely to occur the more frequently the touch panel is sampled
  - The fix:
    - Sanity check every input coordinate pair while looping
    - Count how many coordinate pairs were valid
    - Check there were sufficient valid coordinate pairs (at least 1, and >= 50%)
    - Average only the valid coordinate pairs

Tested on IDF v5.2.1 with a TSC2048 (the TI chip which the XPT2046 cloned)

### CODE REVIEW

File: esp_lcd_touch_xpt2046.c
Comment (atanisoft) [inline esp_lcd_touch_xpt2046.c:164]: I believe it should be available as `tp->config.int_gpio_num`.

Comment (dvosully) [reply esp_lcd_touch_xpt2046.c:164]: You are correct. Amendment incoming.

**Comment by dvosully** [comment]
I have added the check for the PENIRQ active to `xpt2046_read_data`. If the input is high (no touch) then it returns no touch immediately, instead of querying the Z pressure registers. 

It builds and runs normally when `CONFIG_XPT2046_INTERRUPT_MODE` is enabled and `int_gpio_num` is set to `GPIO_NUM_NC`, but I can't test its operation when connected as the PENIRQ is not connected on my board.

**Comment by atanisoft** [comment]
> I can't test its operation when connected as the PENIRQ is not connected on my board.

I'll see if I can test it out, I've got a touch based board (also using the TI part) to see if I can trigger it.

File: esp_lcd_touch_xpt2046.c
Comment (atanisoft) [inline esp_lcd_touch_xpt2046.c:167]: This check seems to be inverted from what `XPT2046_INTERRUPT_MODE` seems to imply (low = touched)

Comment (dvosully) [reply esp_lcd_touch_xpt2046.c]: You are correct. Fixed with a force push.

File: esp_lcd_touch_xpt2046.c
Comment (atanisoft) [inline esp_lcd_touch_xpt2046.c:59]: If you can convert this line to:
```
static const float XPT2046_TEMP0_COUNTS_AT_25C = (599.5 / 2507 * XPT2046_ADC_LIMIT);
```
And move it below the definition of `XPT2046_ADC_LIMIT` I think we will be good to merge.

Comment (atanisoft) [reply esp_lcd_touch_xpt2046.c:59]: also if possible add a comment of where in the datasheet this formula came from.

Comment (dvosully) [reply esp_lcd_touch_xpt2046.c:59]: On it. Should I bump the version or leave that to you?

Comment (atanisoft) [reply esp_lcd_touch_xpt2046.c]: Sure, let's make it 1.0.4 and if you can add a summary to the changelog.md file. After merge I'll tag 1.0.4 and see how people fare with it :)

Comment (dvosully) [reply esp_lcd_touch_xpt2046.c]: Version bumped and changelog updated.

**Review by atanisoft — APPROVED** [review]

File: esp_lcd_touch_xpt2046.c
Comment (atanisoft) [reply esp_lcd_touch_xpt2046.c]: Looks good, thanks!

**Pull request merged**
